Bottom: 532bf023404f4322819d46ecf1933cbb5822237a
Top:    805d7b9361a26941a30210418402d5e94d9d914d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 17:41:50 +0100

hid/common: Provide a helper function for HIDs to use drawing helpers

Gives the following changes to various HIDs:

  batch:   no longer has NOP fill_pcb_polygon and thindraw_pcb_polygon
             routines - should not be an issue, as the drawing calls the
             common helper routines use are still NOPs for the batch HID.

  extents: now has thindraw_polygon implemented - should not be an issue
  gcode:   now has thindraw_polygon implemented - should not be an issue
  gerber:  now has thindraw_polygon implemented - should not be an issue
  nelma:   now has thindraw_polygon implemented - should not be an issue
  png:     now has thindraw_polygon implemented - should not be an issue
  eps:     now has thindraw_polygon implemented - should not be an issue
  ps:      now has thindraw_polygon implemented - should not be an issue


---

diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index e6eaf52..99c2220 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -244,16 +244,6 @@ batch_fill_polygon (hidGC gc, int n_coords, int *x, int *y)
 }
 
 static void
-batch_fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
-{
-}
-
-static void
-batch_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
-{
-}
-
-static void
 batch_fill_rect (hidGC gc, int x1, int y1, int x2, int y2)
 {
 }
@@ -354,6 +344,8 @@ hid_batch_init ()
 {
   memset (&batch_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&batch_hid);
+
   batch_hid.struct_size           = sizeof (HID);
   batch_hid.name                  = "batch";
   batch_hid.description           = "Batch-mode GUI for non-interactive use.";
@@ -379,8 +371,6 @@ hid_batch_init ()
   batch_hid.draw_rect             = batch_draw_rect;
   batch_hid.fill_circle           = batch_fill_circle;
   batch_hid.fill_polygon          = batch_fill_polygon;
-  batch_hid.fill_pcb_polygon      = batch_fill_pcb_polygon;
-  batch_hid.thindraw_pcb_polygon  = batch_thindraw_pcb_polygon;
   batch_hid.fill_rect             = batch_fill_rect;
   batch_hid.calibrate             = batch_calibrate;
   batch_hid.shift_is_pressed      = batch_shift_is_pressed;
diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index b9db654..a9835d0 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -198,3 +198,10 @@ common_thindraw_pcb_polygon (hidGC gc, PolygonType *poly,
   thindraw_contour (gc, poly->Clipped->contours);
   PolygonHoles (poly, clip_box, thindraw_hole_cb, gc);
 }
+
+void
+common_draw_helpers_init (HID *hid)
+{
+  hid->fill_pcb_polygon     = common_fill_pcb_polygon;
+  hid->thindraw_pcb_polygon = common_thindraw_pcb_polygon;
+}
diff --git a/src/hid/common/draw_helpers.h b/src/hid/common/draw_helpers.h
index 3705317..6710b24 100644
--- a/src/hid/common/draw_helpers.h
+++ b/src/hid/common/draw_helpers.h
@@ -2,3 +2,4 @@ void common_fill_pcb_polygon (hidGC gc, PolygonType *poly,
                               const BoxType *clip_box);
 void common_thindraw_pcb_polygon (hidGC gc, PolygonType *poly,
                                   const BoxType *clip_box);
+void common_draw_helpers_init (HID *hid);
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 2b0386c..5ac46ea 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -181,6 +181,8 @@ hid_extents_init (void)
 
   memset (&extents_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&extents_hid);
+
   extents_hid.struct_size         = sizeof (HID);
   extents_hid.name                = "extents-extents";
   extents_hid.description         = "used to calculate extents";
@@ -201,7 +203,6 @@ hid_extents_init (void)
   extents_hid.draw_rect           = extents_draw_rect;
   extents_hid.fill_circle         = extents_fill_circle;
   extents_hid.fill_polygon        = extents_fill_polygon;
-  extents_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   extents_hid.fill_rect           = extents_fill_rect;
 
   initialised = true;
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 3e49ed3..6491afd 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1037,6 +1037,8 @@ hid_gcode_init ()
 {
   memset (&gcode_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&gcode_hid);
+
   gcode_hid.struct_size         = sizeof (HID);
   gcode_hid.name                = "gcode";
   gcode_hid.description         = "G-CODE export.";
@@ -1061,7 +1063,6 @@ hid_gcode_init ()
   gcode_hid.draw_rect           = gcode_draw_rect;
   gcode_hid.fill_circle         = gcode_fill_circle;
   gcode_hid.fill_polygon        = gcode_fill_polygon;
-  gcode_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   gcode_hid.fill_rect           = gcode_fill_rect;
   gcode_hid.calibrate           = gcode_calibrate;
   gcode_hid.set_crosshair       = gcode_set_crosshair;
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index d18cd2a..d5843ef 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1174,6 +1174,8 @@ hid_gerber_init ()
 {
   memset (&gerber_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&gerber_hid);
+
   gerber_hid.struct_size         = sizeof (HID);
   gerber_hid.name                = "gerber";
   gerber_hid.description         = "RS-274X (Gerber) export.";
@@ -1198,7 +1200,6 @@ hid_gerber_init ()
   gerber_hid.draw_rect           = gerber_draw_rect;
   gerber_hid.fill_circle         = gerber_fill_circle;
   gerber_hid.fill_polygon        = gerber_fill_polygon;
-  gerber_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   gerber_hid.fill_rect           = gerber_fill_rect;
   gerber_hid.calibrate           = gerber_calibrate;
   gerber_hid.set_crosshair       = gerber_set_crosshair;
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 84cca05..5af6ed6 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2100,6 +2100,8 @@ hid_gtk_init ()
 
   memset (&ghid_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&ghid_hid);
+
   ghid_hid.struct_size          = sizeof (HID);
   ghid_hid.name                 = "gtk";
   ghid_hid.description          = "Gtk - The Gimp Toolkit";
@@ -2126,8 +2128,6 @@ hid_gtk_init ()
   ghid_hid.draw_rect            = ghid_draw_rect;
   ghid_hid.fill_circle          = ghid_fill_circle;
   ghid_hid.fill_polygon         = ghid_fill_polygon;
-  ghid_hid.fill_pcb_polygon     = common_fill_pcb_polygon;
-  ghid_hid.thindraw_pcb_polygon = common_thindraw_pcb_polygon;
   ghid_hid.fill_rect            = ghid_fill_rect;
 
   ghid_hid.calibrate            = ghid_calibrate;
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 4595b80..c1b12d3 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3802,6 +3802,8 @@ hid_lesstif_init ()
 {
   memset (&lesstif_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&lesstif_hid);
+
   lesstif_hid.struct_size           = sizeof (HID);
   lesstif_hid.name                  = "lesstif";
   lesstif_hid.description           = "LessTif - a Motif clone for X/Unix";
@@ -3828,8 +3830,6 @@ hid_lesstif_init ()
   lesstif_hid.draw_rect             = lesstif_draw_rect;
   lesstif_hid.fill_circle           = lesstif_fill_circle;
   lesstif_hid.fill_polygon          = lesstif_fill_polygon;
-  lesstif_hid.fill_pcb_polygon      = common_fill_pcb_polygon;
-  lesstif_hid.thindraw_pcb_polygon  = common_thindraw_pcb_polygon;
   lesstif_hid.fill_rect             = lesstif_fill_rect;
 
   lesstif_hid.calibrate             = lesstif_calibrate;
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 217a546..6e911ff 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1052,6 +1052,8 @@ hid_nelma_init()
 {
   memset (&nelma_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&nelma_hid);
+
   nelma_hid.struct_size         = sizeof (HID);
   nelma_hid.name                = "nelma";
   nelma_hid.description         = "Numerical analysis package export.";
@@ -1076,7 +1078,6 @@ hid_nelma_init()
   nelma_hid.draw_rect           = nelma_draw_rect;
   nelma_hid.fill_circle         = nelma_fill_circle;
   nelma_hid.fill_polygon        = nelma_fill_polygon;
-  nelma_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   nelma_hid.fill_rect           = nelma_fill_rect;
   nelma_hid.calibrate           = nelma_calibrate;
   nelma_hid.set_crosshair       = nelma_set_crosshair;
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 7c2f461..3919a6e 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1521,6 +1521,8 @@ hid_png_init ()
 {
   memset (&png_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&png_hid);
+
   png_hid.struct_size = sizeof (HID);
   png_hid.name        = "png";
   png_hid.description = "GIF/JPEG/PNG export.";
@@ -1546,7 +1548,6 @@ hid_png_init ()
   png_hid.draw_rect           = png_draw_rect;
   png_hid.fill_circle         = png_fill_circle;
   png_hid.fill_polygon        = png_fill_polygon;
-  png_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   png_hid.fill_rect           = png_fill_rect;
   png_hid.calibrate           = png_calibrate;
   png_hid.set_crosshair       = png_set_crosshair;
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 17c7697..f94175f 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -642,6 +642,8 @@ hid_eps_init ()
 {
   memset (&eps_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&eps_hid);
+
   eps_hid.struct_size         = sizeof (HID);
   eps_hid.name                = "eps";
   eps_hid.description         = "Encapsulated Postscript";
@@ -666,7 +668,6 @@ hid_eps_init ()
   eps_hid.draw_rect           = eps_draw_rect;
   eps_hid.fill_circle         = eps_fill_circle;
   eps_hid.fill_polygon        = eps_fill_polygon;
-  eps_hid.fill_pcb_polygon    = common_fill_pcb_polygon;
   eps_hid.fill_rect           = eps_fill_rect;
   eps_hid.calibrate           = eps_calibrate;
   eps_hid.set_crosshair       = eps_set_crosshair;
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 616f5d5..a417753 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1349,6 +1349,8 @@ hid_ps_init ()
 {
   memset (&ps_hid, 0, sizeof (HID));
 
+  common_draw_helpers_init (&ps_hid);
+
   ps_hid.struct_size        = sizeof (HID);
   ps_hid.name               = "ps";
   ps_hid.description        = "Postscript export.";
